vmdb/app/models/mixins/authentication_mixin.rb

```diff
@@ -92,8 +120,14 @@ def update_authentication(data, options = {})
       end
 
       # Update or create
-      cred = self.authentications.build(:name => "#{self.class.name} #{self.name}", :authtype => type.to_s, :type => "AuthUseridPassword") if cred.nil?
-      cred.userid, cred.password = value[:userid], value[:password]
+      if self.kind_of?(EmsOpenstackInfra) && value[:auth_key]
+        # TODO(lsmola) investigate why build throws an exception, that it needs to be subclass of AuthUseridPassword
```

Comment:
Can it be, that it's because self.authentications already contains one AuthUseridPassword item?

I was not using build method much in the past, definitely not with that much of STI, so I wonder if there is limitation like that.